presto-main/src/main/java/com/facebook/presto/sql/planner/RelationPlanner.java

```diff
@@ -366,6 +372,112 @@ else if (firstDependencies.stream().allMatch(right::canResolve) && secondDepende
         return new RelationPlan(root, analysis.getScope(node), outputSymbols);
     }
 
+    private RelationPlan planJoinUsing(Join node, RelationPlan left, RelationPlan right)
+    {
+        /* Given: l JOIN r USING (k1, ..., kn)
+
+           produces:
+
+            - project
+                    coalesce(l.k1, r.k1)
```

Comment:
is `coalesce` needed? In case any of them is null then join condition would not be met.

Reply:
Yes, an outer join might produce a null on one side. Also, that's how they are define in the spec:

>iii) Let SLCC be a <select list> of <derived column>s of the form:
COALESCE(RVA_i.CJCN_i, RVB_i.CJCN_i) AS CJCN_i
for every i, 1 (one) ≤ i N, in ascending order.